core/templates/components/summary-tile/story-summary-tile.component.spec.ts

```diff
@@ -0,0 +1,448 @@
+    });
+    // Width > 800, to show 3 nodes, which is equal to total number of nodes.
+    spyOn(wds, 'getWidth').and.returnValue(801);
+
+    expect(component.showButton).toBe(undefined);
+
+    component.ngOnInit();
+
+    expect(component.showButton).toBe(false);
+
+    // StorySummary with 5 nodes.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1', 'node2', 'node3', 'node4', 'node5'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1'],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+
+    component.ngOnInit();
+
+    // Will show view all button as number of nodes (5) is more than chapters
+    // displayed (3).
+    expect(component.showButton).toBe(true);
+  });
+
+  it('should show user\'s progress', () => {
+    // StorySummary with 1 node with 0 completed stories.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: [],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+    let circumference = (2 * 20 * Math.PI).toString();
+
+    component.ngOnInit();
+
+    expect(component.strokeDashArrayValues).toBe('');
+    expect(component.completedStrokeDashArrayValues).toBe('0 ' + circumference);
+
+    // StorySummary with 1 node with 1 completed stories.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1'],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+
+    component.ngOnInit();
+
+    expect(component.strokeDashArrayValues).toBe('');
+    expect(component.completedStrokeDashArrayValues).toBe('');
+
+    // StorySummary with 5 node with 3 completed stories.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1', 'node2', 'node3', 'node4', 'node5'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1', 'node2', 'node3'],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+
+    component.ngOnInit();
+
+    // The values are calculated by the same formula as explained above.
```

Comment:
It may be a bit hard to locate the comment being referred to here. It may not be great to rely on the ordering of the tests here -- it may change in the future and the developer may not update the comment. Should we add the comment here again or add a comment at the top of the file after fileoverview so it can be referenced throughout the file?

Reply:
Added a comment here, so that the reader does not need to scroll back again to the top